grouper/repositories/interfaces.py

```diff
@@ -44,3 +64,64 @@ def permissions_for_user(self, user):
     def user_has_permission(self, user, permission):
         # type: (str, str) -> bool
         pass
+
+
+class ServiceAccountRepository(object):
+    """Abstract base class for service account repositories."""
+
+    __metaclass__ = ABCMeta
+
+    @abstractmethod
+    def assign_service_account_to_group(self, username, groupname):
+        # type: (str, str) -> None
+        pass
+
+    @abstractmethod
+    def enable_service_account(self, name):
+        # type: (str) -> None
+        pass
+
+    @abstractmethod
+    def set_service_account_description(self, name, description):
+        # type: (str, str) -> None
+        pass
+
+    @abstractmethod
+    def set_service_account_mdbset(self, name, mdbset):
+        # type: (str, str) -> None
+        pass
+
+
+class UserRepository(object):
+    """Abstract base class for user repositories."""
+
+    __metaclass__ = ABCMeta
+
+    @abstractmethod
+    def add_user_to_group(self, username, groupname, role):
+        # type: (str, str, str) -> None
+        pass
+
+    @abstractmethod
+    def disable_user(self, username):
+        # type: (str) -> None
+        pass
+
+    @abstractmethod
+    def enable_user(self, username):
+        # type: (str) -> None
+        pass
+
+    @abstractmethod
+    def groups_of_user(self, username):
+        # type: (str) -> List[str]
+        pass
+
+    @abstractmethod
+    def mark_disabled_user_as_service_account(self, username):
```

Comment:
This should be a function on the service account repository. (And the comment should be a docstring.)